src/read.c

```diff
@@ -4002,7 +4002,8 @@ static avifResult avifDecoderDecodeTiles(avifDecoder * decoder, uint32_t nextIma
         } else {
             // Non-grid path. Just steal the planes from the only "tile".
             assert(info->tileCount == 1);
-            avifImage * src = decoder->data->tiles.tile[info->firstTileIndex].image;
+            assert(tileIndex == 0);
+            avifImage * src = tile->image;
```

Comment:
At line 3952, the variable `tile` is defined as follows:

```
        avifTile * tile = &decoder->data->tiles.tile[info->firstTileIndex + tileIndex];
```

So if `tileIndex == 0`, then  `decoder->data->tiles.tile[info->firstTileIndex]` is `*tile`.